File: javascript/resistor-color-trio/resistor-color-trio.js
const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white",
];

const colorVal = (color) => {
  const value = COLORS.indexOf(color);
  if (value === -1) {
    throw new Error("invalid color");
  }
  return value;
};

export class ResistorColorTrio {
  constructor([first, second, third]) {
    const zeros = Array(colorVal(third)).fill(0);
    this.value = [colorVal(first), colorVal(second), ...zeros].join("");
  }

  get label() {
    if (this.value > 1000) {
      return `Resistor value: ${this.value / 1000} kiloohms`;
    } else {
      return `Resistor value: ${this.value} ohms`;
    }
  }
}

File: kotlin/resistor-color-trio/src/main/kotlin/ResistorColorTrio.kt
object ResistorColorTrio {
    fun text(vararg input: Color): String {
        val ohms = "${input[0].ordinal}${input[1].ordinal}${"0".repeat(input[2].ordinal)}".toInt()
        return when {
            ohms > 1000_000 -> "${ohms / 1000_000} megaohms"
            ohms > 1000 -> "${ohms / 1000} kiloohms"
            else -> "$ohms ohms"
        }
    }
}
